lib/node_modules/@stdlib/math/base/special/hyp2f1/src/main.c

```diff
@@ -0,0 +1,605 @@
+				q = p * r;
+				y += q;
+				p *= s * ( a+t+d1 ) / ( t+1.0 );
+				p *= ( b+t+d1 ) / ( t+1.0+e );
+				t += 1.0;
+				if ( t > MAX_ITERATIONS ) {
+					*loss = 1.0;
+					return STDLIB_CONSTANT_FLOAT64_NAN;
+				}
+			} while ( y == 0.0 || stdlib_base_abs( q/y ) > EPS );
+
+			if ( id == 0.0 ) {
+				y *= stdlib_base_gamma( c ) / ( stdlib_base_gamma( a ) * stdlib_base_gamma( b ) );
+				goto done;
+			}
+
+			y1 = 1.0;
+			if ( aid != 1.0 ) {
+				t = 0.0;
+				p = 1.0;
+				for ( i = 1; i < aid; i++ ) {
+					r = 1.0 - e + t;
+					p *= s * ( a+t+d2 ) * ( b+t+d2 ) / r;
+					t += 1.0;
+					p /= t;
+					y1 += p;
+				}
+			}
+
+			p = stdlib_base_gamma( c );
+			y1 *= stdlib_base_gamma( e ) * p / ( stdlib_base_gamma( a+d1 ) * stdlib_base_gamma( b+d1 ) );
+			y *= p / ( stdlib_base_gamma( a+d2 ) * stdlib_base_gamma( b+d2 ) );
+			if ( ( (int)aid & 1 ) != 0 ) {
+				y = -y;
+			}
+			q = stdlib_base_pow( s, id );
+			y = ( id > 0.0 ) ? y*q : y1*q;
+			y += y1;
+			goto done;
+		}
+	}
+	// Perform power series if no special cases:
+	y = hys2f1( a, b, c, x, &err );
+
+	// Return the function value:
+	done:
+		*loss = err;
+		return y;
+}
+
+/**
+* Evaluates the Gaussian hypergeometric function.
+*
+* @param a - input value
+* @param b - input value
+* @param c - input value
+* @param x - input value
+* @returns function value
+*
+* @example
+* double v = stdlib_base_gamma( 1.0, 1.0, 1.0, 0.0 );
+* // returns 1.0
+*/
+double stdlib_base_hyp2f1( const double a, const double b, const double c, const double x ) {
+	double negIntCaOrCb;
+	double negIntC;
+	double negIntB;
+	double negIntA;
+	double isIntD;
+	double aid;
+	double err;
+	double ax;
+	double d2;
+	double d1;
+	double id;
+	double ic;
+	double ia;
+	double ib;
+	double t1;
+	double y1;
+	double y2;
+	double q;
+	double r;
+	double p;
+	double e;
+	double s;
+	double d;
+	double y;
+	double i;
+
+	err = 0.0;
+	s = 1.0 - x;
+	d = c - a - b;
+	ax = stdlib_base_abs( x );
+	ia = stdlib_base_round( a );
+	ib = stdlib_base_round( b );
+	id = stdlib_base_round( d );
+	ic = stdlib_base_round( c );
+	negIntA = isNonPositiveInteger( a );
+	negIntB = isNonPositiveInteger( b );
+	negIntC = isNonPositiveInteger( c );
+	isIntD = isInteger( d );
+	t1 = stdlib_base_abs( b-a );
+
+	if ( stdlib_base_is_nan( a ) || stdlib_base_is_nan( b ) || stdlib_base_is_nan( c ) || stdlib_base_is_nan( x ) ) {
+		return STDLIB_CONSTANT_FLOAT64_NAN;
+	}
+
+	if ( x == 0.0 ) {
+		return 1.0;
+	}
+
+	if ( ( a == 0.0 || b == 0.0 ) && c != 0.0 ) {
+		return 1.0;
+	}
+
+	// The transformation for c - a or c - b negative integer (AMS55 #15.3.3)...
+	if ( d <= -1.0 && !( !isIntD && s < 0.0 ) && !( negIntA || negIntB ) ) {
+		return stdlib_base_pow( s, d ) * stdlib_base_hyp2f1( c-a, c-b, c, x );
+	}
+
+	// Check whether the series diverges...
+	if ( d <= 0.0 && x == 1.0 && !( negIntA || negIntB ) ) {
+		goto hypdiv;
```

Comment:
Each of these `goto` statements should be be removed as this sort of misdirection disrupts the linear flow of logic. Yes, they are a language feature, but, in general, they make code harder to reason about.